src/lib/class/Message/Message.cpp

```diff
@@ -0,0 +1,57 @@
+#include "Message.hpp"
+
+/**
+ * Constructor for the Message class.
+ * Parses a single message from the given string.
+ *
+ * @param msg A constant reference to a std::string containing the message to be parsed.
+ *
+ * @return A new instance of the Message class.
+ */
+Message::Message(std::string const &msg)
+{
+	size_t pos = 0;
+	size_t spacePos = 0;
+
+	// Prefix
+	if (msg[pos] == ':')
+	{
+		spacePos = msg.find(' ', pos);
+		this->set_prefix(msg.substr(pos + 1, spacePos - pos - 1));
+		pos = spacePos + 1;
+	}
+
+	// Command
+	spacePos = msg.find(' ', pos);
+	if (spacePos != std::string::npos)
+	{
+		this->set_command(msg.substr(pos, spacePos - pos));
+		pos = spacePos + 1;
+	}
+	else
+	{
+		this->set_command(msg.substr(pos));
+		return;
+	}
+
+	// Params
+	while ((spacePos = msg.find(' ', pos)) != std::string::npos)
+	{
+		if (msg[pos] == ':')
+		{
+			this->add_param(msg.substr(pos + 1));
+			break;
```

Comment:
You'd better `return` instead of `break` at this point, because if you fall in this `if`, that means that you have reached the last argument (starting with a `:`). Doing so will also allows you to remove your following explicit `else`.